src/keccak256_ops.rs

```diff
@@ -0,0 +1,32 @@
+use crate::allocator::{Allocator, NodePtr};
+use crate::cost::check_cost;
+use crate::cost::Cost;
+use crate::op_utils::atom;
+use crate::op_utils::new_atom_and_cost;
+use crate::reduction::Response;
+use sha3::{Digest, Keccak256};
+
+const KECCAK256_BASE_COST: Cost = 50;
+const KECCAK256_COST_PER_ARG: Cost = 160;
+const KECCAK256_COST_PER_BYTE: Cost = 2;
+
+pub fn op_keccak256(a: &mut Allocator, mut input: NodePtr, max_cost: Cost) -> Response {
+    let mut cost = KECCAK256_BASE_COST;
+
+    let mut byte_count: usize = 0;
+    let mut hasher = Keccak256::new();
```

Comment:
For base64 ops you do this in a separate pass, so that you can exit early if the operation is too expensive - is this worth doing here too?

Reply:
the main purpose of the first pass in the base64 functions is to establish the buffer size of the result, since the `base64ct` library only provides constant-time operations, we need to pre-allocate the buffer.